WASM.md

```diff
@@ -0,0 +1,36 @@
+Support for WebAssembly
+=======================
+
+# Summary
+
+Due to the need for a standard library in the build these quantum routines it is not
+possible to compile them as pure no-std WASM however by using wasi and its standard
+library it becomes possible.
```

Comment:
```suggestion
Due to the need for a standard library to build these quantum routines it is not
possible to compile them as pure no-std WASM. However by using wasi and its standard
library, it becomes possible.
```

Why is that, by the way?

Reply:
Its quite hard to resolve all the dependencies and while the routines that the code uses don't really use things from the standard library that would prohibit them running in a sandbox environment (such as no-std) unfortunately the dependency graphs do... which means you end up chasing submitting merge requests to everything upstream until its better supported.

The WASI movement lowers the bar on how much needs to be refactored upstream as it keeps some of the common functions everyone is dependent on.

In a way the very problem is the reason I'm submitting this to your code upstream as if I don't its not my dependency that hurts but its the dependency on the dependencies.

Over time this problem will go away as more and more upstream merges are accepted - once we get a critical mass on WebAssembly WASI compatible libraries then I think the community will move onto "no-std" patches more and more.

Basically for me personally I run two binaries for most of my code, one that compiles on
--target wasm32-unknown-unknown (a.k.a. no-std)

And another on
--target wasm32-wasi

That way I can work on different upstream merges based on the health of the ecosystem as a whole. pqcrypto falls in the wasm32-wasi bucket because its largely still written with the standard library which is just too hard to remove entirely... at the moment at least.